Documentation/contributing/development/debugging.rst

```diff
@@ -4,10 +4,69 @@
     Please use the official rendered version released here:
     https://docs.cilium.io
 
+.. _gs_debugging:
+
 #########
 Debugging
 #########
 
+Attaching a debugger
+--------------------
+
+Cilium comes with a handy set of Makefile targets for quickly deploying
+development builds to a local :ref:`Kind <gs_kind>` cluster. One of them is
+``kind-debug-agent``, generating a container image that wraps the Cilium agent
+with a `dlv <https://github.com/go-delve/delve>`_ invocation, causing the agent
+process to listen for connections from a debugger frontend on port 2345.
+
+To build and push a debug image to your local Kind cluster, run:
+
+.. code-block:: shell-session
+
+    $ make kind-debug-agent
+
+.. note::
+      The image is automatically pushed to the Kind nodes, but running Cilium
```

Comment:
Can you add an action statement here to inform users what they need to do and why it is important that Cilium Pods are not restarted?